### PR TITLE
Update cat plugin to use new JSON cat api

### LINKED ISSUE #9135: prow: dogs win the fighting
When I call my cat, I can't find it. (https://github.com/kubernetes/kubernetes/pull/62582#issuecomment-415313522)

When I call my dog, it appears. (https://github.com/kubernetes/kubernetes/pull/62582#issuecomment-415313654)

Anyway, interesting plugins!

/cc @fejta

### PR DESCRIPTION
**Pull request by ixdy**
The Cat API was recently updated with a completely new and seemingly incompatible API.

The browse/vote interface no longer exists, so `source_url` was removed from the response. It also seems that CATegories are no longer supported. I've removed this functionality from the cat plugin.

On the plus side, it now supports json, which is moderately nicer than XML.

Fixes #9135.

### CODE REVIEW

**Comment by ixdy** [comment]
I've tested this with `-human`.

**Comment by bobcatfish** [comment]
Thanks god 😭

File: prow/plugins/cat/cat_test.go [hunk 1]
```diff
@@ -39,10 +38,7 @@ var human = flag.Bool("human", false, "Enable to run additional manual tests")
 var category = flag.String("category", "", "Request a particular category if set")
```
Comment (fejta) [inline prow/plugins/cat/cat_test.go]: Remove

File: prow/plugins/cat/cat_test.go [hunk 2]
```diff
@@ -354,17 +308,9 @@ func TestCats(t *testing.T) {
 			state:         "open",
 			action:        github.GenericCommentActionCreated,
 			body:          "/meow clothes",
-			shouldComment: true,
+			shouldComment: false,
```
Comment (fejta) [inline prow/plugins/cat/cat_test.go]: Given that this is a regression in ux a better response here would be to treat this as equivalent to `/meow` and leave a comment teaching the person that categories are no longer supported. 

Then after a few months switch to this behavior.

**Comment by fejta** [comment]
thanks!

File: prow/plugins/cat/cat_test.go (hunk 1 above)
Comment (ixdy) [reply prow/plugins/cat/cat_test.go]: done

File: prow/plugins/cat/cat_test.go (hunk 2 above)
Comment (ixdy) [reply prow/plugins/cat/cat_test.go]: cat now **chomps** if you request a category

**Comment by ixdy** [comment]
apparently categories, source URL, voting, etc are still supported, just the docs are wrong. not sure we should deploy this after all.

**Comment by fejta** [comment]
Maybe keep the switch to the json api?

**Pull request merged**